src/LSST/system/TelemetryItem.h

```diff
@@ -114,115 +132,297 @@ class TelemetryItem {
 
-    virtual ~TItemPowerStatusBase() = default;
+    /// Create a shared pointer instance of TItemDouble using `id`, and `defaultVal`, and
+    /// insert it into the map `itMap`.
+    /// @throws TelemetryException if the new object cannot be inserted into the list.
+    static Ptr create(std::string const& id, TelemetryItemMap* tiMap, double defaultVal = 0.0) {
+        Ptr newItem = Ptr(new TItemDouble(id, defaultVal));
+        insert(tiMap, newItem);
+        return newItem;
+    }
 
-    /// Set `_motorVoltage` to `val`.
-    void setMotorVoltage(double val) { _motorVoltage->setVal(val); }
+        /// Create a `TItemDouble` object with identifier `id` and option value `defaultVal`.
+    TItemDouble(std::string const& id, double defaultVal = 0.0) : TItemSimple(id, defaultVal) {}
 
-    /// Set `_motorCurrent` to `val`.
-    void setMotorCurrent(double val) { _motorCurrent->setVal(val); }
+    TItemDouble() = delete;
 
-    /// Set `_commVoltage` to `val`.
-    void setCommVoltage(double val) { _commVoltage->setVal(val); }
+    virtual ~TItemDouble() {}
+};
 
-    /// Set `_commCurrent` to `val`.
-    void setCommCurrent(double val) { _commCurrent->setVal(val); }
+/// This class is used to store a specific telemetry item with a bool value.
+/// Unit tests in tests/test_TelemetryCom
+class TItemBoolean : public TItemSimple<bool> {
+public:
+    using Ptr = std::shared_ptr<TItemBoolean>;
 
-    /// Return the value of `_motorVoltage`.
-    double getMotorVoltage() { return _motorVoltage->getVal(); }
+    /// Create a shared pointer instance of TItemBoolean using `id`, and `defaultVal`, and
+    /// insert it into the map `itMap`.
+    /// @throws TelemetryException if the new object cannot be inserted into the list.
+    static Ptr create(std::string const& id, TelemetryItemMap* tiMap, bool defaultVal = 0.0) {
+        Ptr newItem = Ptr(new TItemBoolean(id, defaultVal));
+        insert(tiMap, newItem);
+        return newItem;
+    }
+
+        /// Create a `TItemBoolean` object with identifier `id` and option value `defaultVal`.
+    TItemBoolean(std::string const& id, double defaultVal = 0.0) : TItemSimple(id, defaultVal) {}
+
+    TItemBoolean() = delete;
+
+    virtual ~TItemBoolean() {}
+};
+
+
+/// This is a TelemetryItem child class for handling vectors, primarily
+/// for the purpose of reading them in and out of json objects.
+/// Unit tests in tests/test_TelemetryCom
+template <class VT>
+class TItemVector : public TelemetryItem {
+public:
+    using Ptr = std::shared_ptr<TItemVector>;
+
+    TItemVector() = delete;
+
+    /// Create a `TItemVector` object with identifier `id` and option value `defaultVal` for all entries.
+    TItemVector(std::string const& id, int size, VT defaultVal = 0) : TelemetryItem(id), _size(size) {
+        for(size_t j=0; j<_size; ++j) {
+            _vals.push_back(defaultVal);
+        }
+    }
+
+    ~TItemVector() override {}
+
+    //// Return a vector copy of the values in this object.
```

Comment:
Based on your coding style, it would be `///` instead of `////`.

Reply:
fixed